Assets/Scripts/MazeGenerator.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System.Collections;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MazeGenerator : MonoBehaviour
```

Comment:
will your code work when width does not equal height?

Reply:
Yes, it will. I also fixed it to work when width is greater than height